### PR TITLE
feat: update component colors to meet minimum AA compliance with WCAG 2.0

### LINKED ISSUE #29: Disabled menu items don't look disabled
Hope this is the correct place to raise this...

With the recent Garden updates on the Talk admin, we have this newly styled dropdown menu.

screen shot 2018-02-21 at 1 29 08 pm

It's used to bulk update numbers that you may have selected.
However, if you have not selected any numbers, all the menu items are disabled and have the grey font colour. 
It's not very clear that they're unclickable, especially as they seem to be a shade darker than the clickable tab links above.

Perhaps we should be hiding the menu if you haven't selected any numbers, but I still think maybe the colours here should be reviewed for disabled text and items.

Willing to provide more context or contribute code if needed :)

cc: @austin94

### PR DESCRIPTION
**Pull request by jzempel**
- [x] **BREAKING CHANGE?**

## Description

Along with sweeping color updates and the removal of "dark mode" styling, note the following breaking changes:

* Avatars – removed `.is-active` and `.is-in` states. Renamed `.is-out` to `.is-disabled` for consistency with other components.
* Tags – remove `.c-tag--light`, `.c-tag--dark`, `.c-tag--error`, `.c-tag--success`, and `.c-tag--warning`. Replaced by `.c-tag--[color]` modifiers.

Added support for:

* `.c-tooltip.is-rtl`
* `.c-tooltip__title`
* `.c-tooltip__paragraph`

## Detail

Closes #29 

## Checklist

* [x] :ok_hand: style updates are Garden Designer approved (add the
  designer as a reviewer)
* [x] :globe_with_meridians: component demo is up-to-date (`yarn start`)
* [x] :white_check_mark: all component states are represented
  (`.is-hovered`, `.is-focused`, etc.)
* [x] :arrow_left: renders as expected with reversed (RTL) direction
* [x] :metal: renders as expected sans Bedrock (`?bedrock=false`)
* [x] :nail_care: provides `custom.css` example for modifying the
  primary accent color
* [x] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

**Comment by jzempel** [comment]
@allisonacs demo is pre-published for review https://garden.zendesk.com/css-components/

File: packages/tags/src/_remove.css [hunk 1]
```diff
@@ -77,6 +75,18 @@
   display: none;
 }
 
+.c-tag.c-tag--blue .c-tag__remove,
+.c-tag.c-tag--green .c-tag__remove,
+.c-tag.c-tag--grey .c-tag__remove,
+.c-tag.c-tag--kale .c-tag__remove,
+.c-tag.c-tag--red .c-tag__remove {
```
Comment (ryanseddon) [inline packages/tags/src/_remove.css]: Not sure if this is better but we could use `:matches()` here e.g.

```css
.c-tag:matches(
  .c-tag--blue, 
  .c-tag--green, 
  .c-tag--grey, 
  .c-tag--kale, 
  .c-tag--red
) :matches(.c-tag__remove) {
  background-image: var(--zd-tag--color__remove-background-image);
} 
```

File: packages/avatars/src/_state.css [hunk 2]
```diff
@@ -2,46 +2,28 @@
 
 /* stylelint-disable no-unsupported-browser-features */
 :root {
-  --zd-avatar-active-background-color: var(--zd-color-malibu);
-  --zd-avatar-in-background-color: var(--zd-color-apple-green);
-  --zd-avatar-out-background-color: var(--zd-color-gainsboro);
-  --zd-avatar-out-filter-url: url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg"><filter id="grayscale"><feColorMatrix values=".3333 .3333 .3333 0 0 .3333 .3333 .3333 0 0 .3333 .3333 .3333 0 0 0 0 0 1 0"/></filter></svg>#grayscale');
-  --zd-avatar-out-opacity: .5;
+  --zd-avatar-disabled-background-color: var(--zd-color-grey-300);
+  --zd-avatar-disabled-filter-url: url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg"><filter id="grayscale"><feColorMatrix values=".3333 .3333 .3333 0 0 .3333 .3333 .3333 0 0 .3333 .3333 .3333 0 0 0 0 0 1 0"/></filter></svg>#grayscale');
```
Comment (ryanseddon) [inline packages/avatars/src/_state.css]: Can you use the svg-load function here?

**Comment by jzempel** [comment]
Cool. Thanks for the thorough review, @allisonacs. I'm going to be dropping a series of commits to address the issues you've raised. I'll ping you when it's ready for a secondary review.

File: packages/tags/src/_remove.css (hunk 1 above)
Comment (jzempel) [reply packages/tags/src/_remove.css]: Huh. TIL and postcss results in the original output.

File: packages/avatars/src/_state.css (hunk 2 above)
Comment (jzempel) [reply packages/avatars/src/_state.css]: `svg-load` requires a file (on a configured path). I don't think it's worth it to get fancy here.

**Pull request merged**